services-directions-models/src/main/java/com/mapbox/api/directions/v5/models/BannerComponents.java

```diff
@@ -501,5 +502,8 @@ public abstract static class Builder {
      * @since 3.0.0
      */
     public abstract BannerComponents build();
+
+    @Nullable
+    abstract Builder unrecognised(Map<String, Object> value);
```

Comment:
we know that unrecognized fields are Strings, this way, let's make `Map<String, String>`

Reply:
it's not only strings. It can also be `JsonObject`, `JsonArray`, etc.

Reply:
why do you want to operate with these fields as JsonObject or JsonArrays? Looks like we only need to serialize/deserialize them. So no need to keep Objects

Reply:
I was actually thinking whether we could consider making this map public (no in builder but in the resulting type), so that if someone would like to access the unrecognized fields, they don't need to serialize the type first. I believe this will be needed soon (for example for experimental EV properties).

Reply:
@LukasPaczos Experimental EV and other options are allocated in RouteOptions. We don't extend RouteOptions with Unrecognized fields I think 🤔 Suppose, it's separate initiative, not really connected with this PR

Reply:
`RouteOptions` extends `DirectionsJsonObject` so it already has unrecognised fields